uiTests/src/test/scala/com/virtuslab/gitmachete/uitest/GitMacheteExtension.scala

```diff
@@ -0,0 +1,113 @@
+import org.virtuslab.ideprobe.Extensions._
+import org.virtuslab.ideprobe.IdeProbeFixture
+import org.virtuslab.ideprobe.dependencies.Plugin
+import org.virtuslab.ideprobe.robot.RobotPluginExtension
+
+trait GitMacheteExtension extends RobotPluginExtension { this: IdeProbeFixture =>
+
+  private val machetePlugin: Plugin = {
+    val cwd = Paths.get(System.getProperty("user.dir"))
+    val repositoryRoot = cwd.getParent
+    val distedPluginsDir = repositoryRoot.resolve("build/distributions")
+    val latestFileInDistDir = distedPluginsDir.toAbsolutePath.toFile.listFiles().maxBy(_.lastModified())
+    Plugin.Direct(latestFileInDistDir.toURI)
+  }
+
+  registerFixtureTransformer(_.withPlugin(machetePlugin))
+  registerFixtureTransformer(_.withAfterIntelliJStartup((_, intelliJ) => intelliJ.machete.runJs("ide.configure(/* enableDebugLog */ false)")))
+
+  private val rhinoCodebase = {
+    def loadScript(baseName: String) = {
+      Paths.get(getClass.getResource(s"/$baseName.rhino.js").toURI).content()
+    }
+
+    Seq("common", "ide", "project").map(loadScript).mkString
+  }
+
+  implicit class MacheteExtensions(intelliJ: RunningIntelliJFixture) {
+    object machete {
+      def refreshModelAndGetRowCount(): Int = {
+        callJs("project.refreshGraphTableModel().getRowCount()")
+      }
+
+      def getHashOfCommitPointedByBranch(branch: String): String = {
+        callJs(s"project.getHashOfCommitPointedByBranch('$branch')")
+      }
+
+      def getDiffOfWorkingTreeToHead(): Seq[String] = {
+        callJs[util.ArrayList[String]]("project.getDiffOfWorkingTreeToHead()").asScala
+      }
+
+      def checkoutBranch(branch: String): Unit = {
+        runJs(s"project.checkoutBranch('$branch')")
+        intelliJ.probe.awaitIdle()
+      }
+
+      def fastForwardParentToMatchBranch(branch: String): Unit = {
+        runJs(s"project.fastForwardParentToMatchBranch('$branch')")
+        intelliJ.probe.awaitIdle()
+      }
+
+      def fastForwardParentToMatchCurrentBranch(): Unit = {
+        runJs(s"project.fastForwardParentToMatchCurrentBranch()")
+        intelliJ.probe.awaitIdle()
+      }
+
+      def pullCurrentBranch(): Unit = {
+        runJs(s"project.pullCurrentBranch()")
+        intelliJ.probe.awaitIdle()
+      }
+
+      def pullBranch(branch: String): Unit = {
+        runJs(s"project.pullBranch('$branch')")
+        intelliJ.probe.awaitIdle()
+      }
+
+      def resetCurrentBranchToRemote(): Unit = {
+        runJs(s"project.resetCurrentBranchToRemote()")
+        intelliJ.probe.awaitIdle()
+      }
+
+      def resetBranchToRemote(branch: String): Unit = {
+        runJs(s"project.resetBranchToRemote('$branch')")
+        intelliJ.probe.awaitIdle()
+      }
+
+      def getCurrentBranchName(): String = {
+        callJs[String]("project.getCurrentBranchName()")
+      }
+
+      def assertWorkingTreeIsAtHead(): Unit = {
+        Assert.assertEquals(Nil, getDiffOfWorkingTreeToHead())
```

Comment:
Nit: maybe `Seq.empty` for more clarity, `Nil` is pretty unclear for non-Scala ppl

Reply:
(esp. given that there's also `null`, `Null`, `None`, `Nothing` etc.)